### ISSUE TITLE
HumHub fully translated to Russian!

### ISSUE THREAD

**Issue by hide-me** [humhub/humhub#4041]
Again hello everyone!

I made a **full translation of HumHub into Russian**. All translation files are finished. The previous partial translation was double-checked, changed, bugs, and typos corrected. The translation was created based on my glossary of terms, so the same functions/features have the same name, no matter in which part of HubHum site they are.

**For localization was taken version 1.5.0.**

[humhub-russian.zip](https://github.com/humhub/humhub/files/4522089/humhub-russian.zip)

_I hope you appreciate the work I have done.
Thank you. I will wait for the answer here in the comments._

**Comment by githubjeka** [comment]
Term `space` trasform to `community`. 
Like transoform '❤'
In some places, words have different semantics.

In general, good work, can be framed as a third-party module, but not officially adopted.

**Comment by hide-me** [comment]
> Term `space` trasform to `community`.
> Like transoform '❤'

Yes, because you know, `space` in Russian we usually use with meaning `area` or `field`. So why is it a problem to use `community`?? In VK (Russian social network) you are also working with `community` not `space`. So it's good to change.

Next. About like. Why using this emoji is bad? You also know, in the previous translation if you liked a post - you see the string with meaning like this: **dislike (3)**

The previous translation was not complete, with mistakes and typos. All my _transformations_ the same in every place of the system and provide a better experience for all users and admins.

**Comment by luke-** [comment]
Some terms like "Space" should not be translated.  

With other terms like Like or Follow I am not sure what is the best approach here.

But we should definitely find a consistent solution here.

**Comment by hide-me** [comment]
> Some terms like "Space" should not be translated.

Sorry, what you mean? In Russain, we usually use "Space" with a different meaning. So that why I changed and adapted this theremin. Again - for Russian users, it's pretty easy, cuz we using this variant in Vk (social network).

**Comment by luke-** [comment]
Please don't misunderstand me, I and also the entire HumHub community really appreciate your efforts. But you should also allow comments or a discussion without being immediately offended.

I do not speak Russian myself and therefore cannot comment on the translation at all.
But for us in Humhub `Space` is an established term - the central component in HumHub.

Space is such a generic construct and there are so many different possibilities in every language - it's also depends on the specific use of HumHub...

But if it should be unacceptable in one language (e.g. by a completely different meaning) I will not prevent myself from finding an alternative for it.

**Comment by hide-me** [comment]
> But if it should be unacceptable in one language (e.g. by a completely different meaning) I will not prevent myself from finding an alternative for it.

It is NOT unacceptable, but the very unusual meaning of this word (space) for Russian users. It's like a situation when I use **"categories of users"** instead of **users group** or **users role**. It is understandable, but strange and confusing.



> But you should also allow comments or a discussion without being immediately offended.

Yep, sorry. Let's try to wait for an answer from @githubjeka or someone else who knows the Russian language very well.

**Comment by luke-** [comment]
By the way, the word `Space` is not common here in German either.
But that's the thing about HumHub, you create Spaces...

Of course there are different words for spaces/rooms/workgroups/classes/... in the various languages/use cases.

**Comment by hide-me** [comment]
> By the way, the word `Space` is not common here in German either.

And it does not confuse or disorient ordinary/newbie users? Because I'm trying to make correct and comfortable translations for users, not because I don't like `Space` definition in Russian or something.

I also finished translation for polls module here, in Github, please check my pull requests in the module repo. Hope you will like it. This translation does not contain `Space` word at all 😆 
Sorry for multiple pull requests, I'm working using the web interface of Github 😼

**Comment by hide-me** [comment]
So, what are you decided?

**Comment by buddh4** [comment]
In the future we should allow overwriting such terms like:

```
Yii::t('base', 'My name is {space}', [
    'space' => Yii::t('base', 'Space')
]);
```

```
Yii::t('base', 'Our name is {spaces}', [
    'spaces' => Yii::t('base', 'Spaces')
]);
```

Or perhaps set the space parameter automatically.

What do you think @luke- ?

**Comment by hide-me** [comment]
Sounds really good!

**Comment by luke-** [comment]
@buddh4 It is often not that easy to simply replace just a single word. There is a singular and plural version and e.g. in German also articles (e.g. der Bereich vs. die Gruppe).

But if we find a clean solution and a contributor for this, we can make the term "space" configurable from my side.

**Comment by hide-me** [comment]
Any news on my topic?

**Comment by buddh4** [comment]
Sorry, this currently does not have any priority, perhaps just [overwrite the translation](https://docs.humhub.org/docs/admin/translations#overwrite-translation-messages) for now.

**Comment by MasterHans** [comment]
Hello everyone!
I hope you doing fine.

With full respect to the author of this translation I want to give you my opinion about **"Space"** and **"Likes"**.

1) When I saw the word **Space** I thought about **outer space**, **rocket ships**,  **starship troopers**, and sci-fi. About **cosmos**.
I think that **community** - meaning a group of people nearby and their connections and relatives,  
That does not completely fit **Space** definition of humhub because spaces contain **polls** - people's opinions, but not only people themselves. So you could provide surveys based on it, etc. 
Thus **Space** is **Ok**. Current translations as "Пространстов" - totally alright with me.

2) About emoji - this is a design (style) question, not translation. If you going to set emoji for a **like** you must set emoji for a ** comment** etc. So, the current translation of **like** is alright with me as well.

Thank you
Best regards
Alex

**Comment by buddh4** [comment]
See https://github.com/humhub/humhub/issues/2415 regarding a reaction feature request.

**Comment by githubjeka** [comment]
https://github.com/humhub/humhub/pull/4405 

Was it merged by accident or did you accept the term "communities"?

**Comment by luke-** [comment]
@MasterHans Rather an accident, wasn't it? Space would already be the preferred choice.

**Comment by hide-me** [comment]
Thanks for everyone who participate in this discussion. The time, users, modern UI standarts, e t. c. made my version of translation comfy and official.



Please close the issue as solved.
